### PR TITLE
Add `List.flatMap` (fixes #352)

### LINKED ISSUE #352: [Stdlib] Add List.flatMap
Here's `flatMap` signature:

```grain
flatMap : (a -> List<b>, List<a>) -> List<b>
```

### PR DESCRIPTION
**Pull request by ohana54**

### CODE REVIEW

File: compiler/test/stdlib/list.test.gr [hunk 1]
```diff
@@ -41,6 +41,11 @@ let multiply = (n, i) => n * i
 assert mapi(multiply, numbers) == [0, 3, 8]
 assert mapi(multiply, []) == []
 
+# List.flatMap
+
+assert flatMap(x => [x + 1], list) == [2, 3, 4]
```
Comment (ospencer) [inline compiler/test/stdlib/list.test.gr:46]: Would maybe be good to also test having multiple items in the list:
```suggestion
assert flatMap(x => [x + 1], list) == [2, 3, 4]
assert flatMap(x => [x, x + 1], list) == [1, 2, 2, 3, 3, 4]
```

**Review by ospencer — CHANGES_REQUESTED** [review]
Looks great! Just one idea about an additional test.

File: compiler/test/stdlib/list.test.gr (hunk 1 above)
Comment (ohana54) [reply compiler/test/stdlib/list.test.gr:46]: No idea how I missed this case 🤦

**Review by ospencer — APPROVED** [review]
Nice! 🎉

**Pull request merged**